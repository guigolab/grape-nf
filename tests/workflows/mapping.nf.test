nextflow_workflow {

    name "Test Workflow mapping"
    script "workflows/mapping.nf"
    workflow "mapping"

    tag "workflow"

    test("Should run STAR mapping sorting with sambamba") {

        when {
            params {
                mappingTool  = "STAR"
                quantificationTool = "RSEM"
                sortBamTool = "sambamba"
                sjOverHang = 100
                maxMismatches = 4
                maxMultimaps = 10
            }
            workflow {
                """
                input[0] = file("${baseDir}/data/genome.fa.gz")
                input[1] = file("${baseDir}/data/annotation.gtf.gz")
                input[2] = Channel.from(
                    [
                        [ "sample1", "test1", file("${baseDir}/data/test1_*.fastq.gz").sort(), "fastq", ["fqRd1", "fqRd2"], '']
                    ]
                )
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.genomeAlignments,
                workflow.out.genomeAlignmentsIndices,
                workflow.out.transcriptomeAlignments,
                workflow.out.junctions
            ).match()
        }

    }

    test("Should run STAR mapping sorting with samtools") {

        when {
            params {
                mappingTool  = "STAR"
                quantificationTool = "RSEM"
                sortBamTool = "samtools"
                sjOverHang = 100
                maxMismatches = 4
                maxMultimaps = 10
            }
            workflow {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                input[2] = Channel.from(
                    [
                        [ "sample1", "test1", file("${baseDir}/data/test1_*.fastq.gz").sort(), "fastq", ["fqRd1", "fqRd2"], '']
                    ]
                )
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.genomeAlignments,
                workflow.out.genomeAlignmentsIndices,
                workflow.out.transcriptomeAlignments,
                workflow.out.junctions
            ).match()
        }

    }

}
