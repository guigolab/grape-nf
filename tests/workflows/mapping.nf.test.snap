{
    "Should run STAR mapping sorting with samtools": {
        "content": [
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toGenome_sorted.bam:md5,162f23dcd62197a2c1409294ef99e690",
                    "bam",
                    "GenomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toGenome_sorted.bam.bai:md5,afa4023ac7c17c64aa67d833cf9276e3",
                    "bam",
                    "GenomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toTranscriptome.bam:md5,e6a72cca10b1ac084ff2ae4f572987b8",
                    "bam",
                    "TranscriptomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "SJ.out.tab:md5,ec651e0f3bc8338345aa15c8f74038c2",
                    "tsv",
                    "STARjunctions",
                    true
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-17T15:01:17.646174"
    },
    "Should run STAR mapping sorting with sambamba": {
        "content": [
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toGenome_sorted.bam:md5,a7a3d9267bd0c57c568857e729219a0e",
                    "bam",
                    "GenomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toGenome_sorted.bam.bai:md5,8f0bdb7f22db60c70983eb6a56e66b3c",
                    "bam",
                    "GenomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "sample1_m4_n10_toTranscriptome.bam:md5,e6a72cca10b1ac084ff2ae4f572987b8",
                    "bam",
                    "TranscriptomeAlignments",
                    true
                ]
            ],
            [
                [
                    "sample1",
                    "test1",
                    "SJ.out.tab:md5,ec651e0f3bc8338345aa15c8f74038c2",
                    "tsv",
                    "STARjunctions",
                    true
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-17T15:00:46.24496"
    }
}